src/engine/Join.h

```diff
@@ -54,6 +54,23 @@ class Join : public Operation {
 
   virtual float getMultiplicity(size_t col) override;
 
+  /**
+   * @brief Joins IdTables dynA and dynB on join column jc2, returning
+   * the result in dynRes. Creates a cross product for matching rows
+   **/
+  template <int L_WIDTH, int R_WIDTH, int OUT_WIDTH>
```

Comment:
I'm a little confused here. The template itself is in the `.cpp` but having the declaration in the header seems to be enough? Isn't this slightly fragile as it depends on all specializations being instantiated in the `Join.cpp` or am I missing something?